### PR TITLE
Added check whether writer is opened before trying to close book. Fixes #392.

### LINKED ISSUE #392: Xlsx writer closes file even if already closed.
Hi,

Encountered an issue when trying to call close() method twice:
> [Box\Spout\Common\Exception\IOException]                       
>   Cannot perform I/O operation outside of the base folder: /tmp

It seems that writer checks for book to be present, even if isWriterOpened flag is false:
```
/**
     * Closes the writer, preventing any additional writing.
     *
     * @return void
     */
    protected function closeWriter()
    {
        if ($this->book) {
            $this->book->close($this->filePointer);
        }
    }
```

I don't have time to prepare PR now, so I suggest adding  `&& $this->isWriterOpened` condition in 
[src/Spout/Writer/XLSX/Writer.php:128](https://github.com/box/spout/blob/master/src/Spout/Writer/XLSX/Writer.php#L128).

Thanks in advance!
P.S. Let me know if you need a PR for this, I'll do my best to create one in a few days.
P.P.S. I tried to implement some cleanup / rollback functionality. I suppose creating a public method with `closeAndAttemptToCleanupAllFiles()` code inside would be nice ;)

### PR DESCRIPTION
**Pull request by Zorato**
This PR fixes #392.

### CODE REVIEW

**Comment by boxcla** [comment]
Hi @Zorato, thanks for the pull request. Before we can merge it, we need you to sign our Contributor License Agreement. You can do so electronically here: http://opensource.box.com/cla

Once you have signed, just add a comment to this pull request saying, "CLA signed". Thanks!

**Comment by Zorato** [comment]
CLA signed

**Comment by boxcla** [comment]
Verified that @Zorato has just signed the CLA. Thanks, and we look forward to your contribution.

File: src/Spout/Writer/XLSX/Writer.php [hunk 1]
```diff
@@ -125,7 +125,7 @@ protected function getDefaultRowStyle()
      */
     protected function closeWriter()
     {
-        if ($this->book) {
+        if ($this->book && $this->isWriterOpened) {
```
Comment (adrilo) [inline src/Spout/Writer/XLSX/Writer.php]: This addition should be moved to the `close()` function inside `AbstractWriter` so it's applied to all writers and not only the XLSX one.

File: tests/Spout/Writer/XLSX/WriterTest.php [hunk 2]
```diff
@@ -491,6 +491,19 @@ public function testGeneratedFileShouldHaveTheCorrectMimeType()
     }
 
     /**
+     * @return void
+     */
+    public function testCloseIsNotAppliedWhenWriterIsNotOpened()
+    {
+        /** @var \Box\Spout\Writer\XLSX\Writer $writer */
+        $writer = WriterFactory::create(Type::XLSX);
```
Comment (adrilo) [inline tests/Spout/Writer/XLSX/WriterTest.php]: I would simplify this test this way:

```
/** @var \Box\Spout\Writer\XLSX\Writer $writer */
$writer = WriterFactory::create(Type::XLSX);
$writer->close();

$writer->openToFile('test_double_close_calls.xlsx');
$writer->close();
$writer->close();
```

**Comment by adrilo** [comment]
Closing this PR as I recreated it here: #402 (I'd like to merge it asap to release a new version).

**Pull request closed without merging**